### ISSUE TITLE
CollisionCapsule - CollisionPolygon intersection test bug

### ISSUE THREAD

**Issue by rasmaxim** [panda3d/panda3d#906]
I've encountered what seems to be a bug in collision detection.

Way to reproduce: 

1. Create a collision node with a vertically oriented CollisionCapsule solid, add it to the collision traverser
2. Create a collision node with a horizontally oriented CollisionPolygon some units above the capsule, but not intersecting it

Expected behaviour: no intersection detected
Actual behaviour: intersection is detected, debug log prints:
```
:collide(debug): intersection detected from render/capsule/capsule_collider into render/quad
```

With a CollisionHandlerPusher, there's more info:

```
:collide(debug): intersection detected from render/capsule/capsule_collider into render/quad
:collide(debug): Shove on render/capsule/capsule_collider from render/quad: 0 0 1 times 16
:collide(debug): Net shove on render/capsule/capsule_collider is: 0 0 16
```

16 is the distance between capsule top and the quad. If the distance is more than 16, the collision is not detected.

This value is not constant, it depends on the quad area. The larger the area, the larger the distance.
Also, the larger the area, the larger the distance along the XY plane the collision is detected. Here's the video explaining this part better:  https://youtu.be/W3e7vbNJLnw

Here's code to reproduce: https://gist.github.com/rasmaxim/32dcd54a3fc48e2051ace2d106ebbabc

I don't know if it is a bug, but other solids (at least Box and Sphere) don't behave this way.

**Comment by rdb** [comment]
I'm unable to reproduce this using the code provided.  Can you verify whether it's still happening with the latest master version?

**Comment by rasmaxim** [comment]
This seems to be fixed in 1.11.0.dev1697.
